### ISSUE TITLE
[BUG] "/bin/sh: 1: set: Illegal option -o pipefail"

### ISSUE THREAD

**Issue by dirkvdplas** [thorian93/main#15]
**Describe the bug**
[]
Ubuntu upgrade fails with following message 

(fatal: [elastic-eset-dns-dev]: FAILED! => {"changed": false, "cmd": "set -o pipefail;\nneedrestart -b | grep NEEDRESTART-SVC | cut -d ' ' -f2\n", "delta": "0:00:00.004343", "end": "2024-03-27 09:14:38.919593", "msg": "non-zero return code", "rc": 2, "start": "2024-03-27 09:14:38.915250", "stderr": "/bin/sh: 1: set: Illegal option -o pipefail", "stderr_lines": ["/bin/sh: 1: set: Illegal option -o pipefail"], "stdout": "", "stdout_lines": []})

Execution outside ansible also fails:
user $ /bin/sh -c "set -o pipefail;\nneedrestart -b | grep NEEDRESTART-SVC | cut -d ' ' -f2"
/bin/sh: 1: set: Illegal option -o pipefail

**Component Name**

e.g. `upgrade`

**Ansible Version**
ansible [core 2.15.4]
  config file = None
  configured module search path = ['/home/test-user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/pipx/venvs/ansible-core/lib/python3.10/site-packages/ansible
  ansible collection location = /home/test-user/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/pipx_bin/ansible
  python version = 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0] (/opt/pipx/venvs/ansible-core/bin/python)
  jinja version = 3.1.2
  libyaml = True

```


**Environment**
<!-- Provide all relevant information below, e.g. control node OS versions, etc. -->

PRETTY_NAME="Ubuntu 22.04.4 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.4 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy

**Comment by thorian93** [comment]
Hi @dirkvdplas and thanks for reporting this!

I can see, that the shell for your user running Ansible on your Ubuntu seems to be `/bin/sh`, which is not the default to my knowledge. All the Ubuntus I had hands on use `/bin/bash` for their users.  `/bin/sh` does not support the option used in the role, that is expected. Hence, the role forces the use of `/bin/bash`, which seems to be overridden. Is there a reason, why you are using `/bin/sh`?

**Comment by dirkvdplas** [comment]
If I am correct /bin/sh is default shell for all ansible shell tasks. 
I was able to fix the error by adding "args: executable: /bin/bash" to the task.
Then I ran into other issue, because at this point there were no services which required restart, which caused the task to fail. I have added ignore_errors: true

- name: "Check for Service Restarts."
  ansible.builtin.shell: |
    set -o pipefail
    needrestart -b | grep NEEDRESTART-SVC | cut -d ' ' -f2
  args:
    executable: /bin/bash
  register: service_restarts_apt
  ignore_errors: true
  changed_when: "service_restarts_apt.stdout_lines | length"

**Comment by thorian93** [comment]
I stand corrected. That is why my shell tasks should always set the `executable` arg. But that was broken - for some time apparently. I just fast-tracked the next release to publish the fixes I already made: [1.0.0](https://github.com/thorian93/main/releases/tag/v1.0.0)

Feel free to reopen the issue, if the problem is not solved and thanks for reporting!
